### PR TITLE
TLS Support

### PR DESCRIPTION
**Pull request by domodwyer**
Adds support for connecting to SMTP hosts over a TLS-enabled connection.

The `NewWithTLS()` method dials to the SMTP endpoint using TLS, and accepts a [`tls.Config`](https://golang.org/pkg/crypto/tls/#Config) to allow the caller to control the protocol parameters.

---

@darnmason if you have the time, I'd love to know if this would work for your use case (or any other comments!)

### CODE REVIEW

File: mailyak.go [hunk 1]
```diff
@@ -34,47 +43,86 @@ type MailYak struct {
...
+		//nolint:gosec // Maximum compatability but please use TLS >= 1.2
+		tlsConfig = &tls.Config{
+			ServerName: hostName,
+		}
+	}
+
+	// Construct a default MailYak instance
+	m := New(host, auth)
+
+	// Initialise the TLS sender with the (potentially nil) TLS config,
```
Comment (darnmason) [inline mailyak.go:107]: slightly misleading comment, as it's not nil at this point, and the `newSenderWithExplicitTLS` method requires that it be non nil

**Review by darnmason — DISMISSED** [review]
Awesome, looks perfect!

File: mailyak.go (hunk 1 above)
Comment (domodwyer) [reply mailyak.go:107]: Great catch - I wound up refactoring this and it got left behind

Thanks!

**Comment by domodwyer** [comment]
Thanks for your help @darnmason - much appreciated :) 

I will cut a release shortly, but there's one more change I want to get in first!

**Comment by domodwyer** [comment]
Done! This is now available in the [v3.3.0](https://github.com/domodwyer/mailyak/releases/tag/v3.3.0) release.

Thanks again :)

**Pull request merged**